### ISSUE TITLE
[FR] multidim squeeze

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#30702]
Previous FR: https://github.com/pytorch/pytorch/issues/1951

A PR (https://github.com/pytorch/pytorch/pull/4692) that attempted to implement https://github.com/pytorch/pytorch/issues/1951 was closed because 
> The behavior of this feature is very non-trivial, e.g., when negative and positive indices are mixed. I'm not sure it solves more problems that it adds anymore.

I would argue against that point, particularly for `squeeze`.

For `squeeze`, multidim is perfectly intuitive and defined, i.e., just squeezing the current dimensions the tuple refers. In fact, using a chain of squeeze (as suggested by the comment of https://github.com/pytorch/pytorch/pull/4692) is rather unintuitive. 

If one has a tensor that has size `1` at dims `2` and `3`, which I want to squeeze, it is more intuitive to write 
```
t.squeeze((2, 3))
```
than to write
```
t.squeeze(2).squeeze(2)
```

**Comment by zou3519** [comment]
I agree that it makes sense for squeeze and flatten to have these APIs. I would be happy to review PRs that enable this.

**Comment by vadimkantorov** [comment]
Btw `unsqueeze_multiple` already exists somewhere in PyTorch C++ code: https://github.com/pytorch/pytorch/blob/877c96cddfebee00385307f9e1b1f3b4ec72bfdc/tools/autograd/templates/Functions.cpp#L173

**Comment by vadimkantorov** [comment]
`squeeze_multiple` already exists [somewhere in PyTorch](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/ReduceOps.cpp#L354)

`unsqueeze_multiple` also [exists](https://github.com/pytorch/pytorch/blob/68f378210666c079326139b6557e8a601fc89ebf/tools/autograd/templates/Functions.cpp#L173)

**Comment by vadimkantorov** [comment]
I recently had a practical usecase for unsqueeze_multiple, had to implement it like this:
`unsqueeze_multiple = lambda x, dim, k: functools.reduce(lambda acc, _: acc.unsqueeze(dim), range(k), x)` (https://gist.github.com/vadimkantorov/ba3d3d099e403f2703facd2765625f27)

**Comment by SsnL** [comment]
Oh, so for unsqueeze_multiple, it is just inserting multiple dims at the same location, right? @vadimkantorov

**Comment by vadimkantorov** [comment]
Yup!

**Comment by vadimkantorov** [comment]
A recent related example: https://github.com/pytorch/pytorch/issues/94750

**Comment by vadimkantorov** [comment]
It appears that multidim squeeze is merged now :)
